src/test/java/org/benetech/servicenet/adapter/icarol/AbstractDataAdapterDocumentUploadTest.java

```diff
@@ -0,0 +1,144 @@
+
+import static org.benetech.servicenet.adapter.AdapterTestsUtils.mockEndpointWithBatch;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.mockserver.integration.ClientAndServer.startClientAndServer;
+
+@RunWith(SpringRunner.class)
+@SpringBootTest(classes = ServiceNetApp.class)
+@DirtiesContext(classMode = DirtiesContext.ClassMode.BEFORE_EACH_TEST_METHOD)
+public class AbstractDataAdapterDocumentUploadTest {
+
+    private static final String EDEN_PROVIDER = "Eden";
+    private static final String ICAROL_CATALOG = "icarol/";
+
+    @Mock
+    private HttpClient client;
+
+    @Mock
+    private DocumentUploadRepository documentUploadRepository;
+
+    @Mock
+    private DocumentUploadMapper documentUploadMapper;
+
+    @Mock
+    private DataImportReportService dataImportReportService;
+
+    @Mock
+    private UserService userService;
+
+    @Mock
+    private MongoDbService mongoDbService;
+
+    @Mock
+    private ApplicationContext applicationContext;
+
+    @Autowired
+    @InjectMocks
+    private DocumentUploadServiceImpl documentUploadService;
+
+    @Autowired
+    private EntityManager em;
+
+    private static MockServerClient mockServer = startClientAndServer(1080);
+
+    @Before
+    public void setUp() throws IOException {
+        MockitoAnnotations.initMocks(this);
+
+        String allIds = AdapterTestsUtils.readResourceAsString(ICAROL_CATALOG + "ids.json");
+        ICarolComplexResponseElement data = getAllIdsInBatches(allIds);
+
+        mockEndpointWithBatch(data.getProgramBatches().get(0), ICAROL_CATALOG + "programs.json", mockServer);
+        mockEndpointWithBatch(data.getSiteBatches().get(0), ICAROL_CATALOG + "sites.json", mockServer);
+        mockEndpointWithBatch(
+            data.getServiceSiteBatches().get(0), ICAROL_CATALOG + "serviceSites.json", mockServer);
+        mockEndpointWithBatch(data.getAgencyBatches().get(0), ICAROL_CATALOG + "agencies.json", mockServer);
+
+        em.clear();
+    }
+
+    @AfterClass
+    public static void stopServer() {
+        mockServer.stop();
+    }
+
+    @Test
+    public void shouldSaveAllIdsDocTest() throws IOException {
+        String allIds = AdapterTestsUtils.readResourceAsString(ICAROL_CATALOG + "ids.json");
+        DataImportReport report = new DataImportReport().startDate(ZonedDateTime.now());
+
+        DocumentUploadDTO document = documentUploadService.uploadApiData(allIds, EDEN_PROVIDER, report);
+
+        assertNotNull(document.getOriginalDocumentId());
+    }
+
+    @Test
+    public void shouldCreateOrganizationsTest() throws IOException {
+        String allIds = AdapterTestsUtils.readResourceAsString(ICAROL_CATALOG + "ids.json");
+        DataImportReport report = new DataImportReport().startDate(ZonedDateTime.now());
+
+        documentUploadService.uploadApiData(allIds, EDEN_PROVIDER, report);
+
+        assertEquals(Integer.valueOf(2), report.getNumberOfCreatedOrgs());
```

Comment:
is 'Integer.valueOf' necessary?

Reply:
yes, because I can compare two objects or two primitives